content/guides/use-case/wiremock.md

```diff
@@ -0,0 +1,272 @@
+
+  ![Diagram showing the architecture of WireMock in development ](./images/wiremock-arch.webp)
+
+  * The API Request is sent from the Node.js backend to the External AccuWeather API, which then returns the API Response.
+  * Alternatively, when using WireMock, a Mocked API Request is sent to the WireMock server, and a Mocked API Response is returned.
+
+  In this guide, you’ll learn how to:
+
+  * Set up a non-containerized Node app to route requests to the containerized WireMock 
+  * Set up a non-containerized Node app to fetch data from the external AccuWeather API.
+
+  The application can be accessed at [dockersamples/wiremock-node-docker](https://github.com/dockersamples/wiremock-node-docker).
+
+
+## Set up a non-containerized Node app to route requests to a containerized WireMock
+
+Let’s set up a non-containerized Node app to route requests to the WireMock container instead of the external AccuWeather API. Assuming that Node is already installed on your system, follow the steps:
+
+1. Navigate to the `accuweather-api` directory
+
+   Make sure you're in the directory where your `package.json` file is located.
+
+2. Set the environment variable.
+
+   Open `.env` file placed under `accuweather-api/` directory. Remove the old entries and ensure that it just contains the following single line.
+ 
+   ```plaintext
+   API_ENDPOINT_BASE=http://localhost:8080
+   ```
+
+   This will tell your Node.js application to use the WireMock server for API calls. 
+
+3. Start the Node server
+
+   Before you start the Node server, ensure that you have already installed the node packages listed in the package.json file by running `npm install`. 
+
+   ```console
+   npm install 
+   npm run start
+   ```
+ 
+   You should see the following output:
+
+    ```plaintext
+    > express-api-starter@1.2.0 start
+    > node src/index.js
+
+    API_ENDPOINT_BASE: http://localhost:8080
+    ..
+    Listening: http://localhost:5000
+    ```
+
+   The output indicates that your Node application has successfully started. It highlights the project name and the `index.js` file as the application's entry point. The `index.js` file initiates the `getWeather.js` file located within the `src/api/` directory by utilizing the `require` function to import the `api` module and subsequently invoking the `getWeather` function.
+
+   The `getWeather.js` file plays a pivotal role within your Node.js application. The code commences by incorporating the `dotenv` module, which instructs the application to load environment variables from a `.env` file. Depending on the value of the `API_ENDPOINT_BASE` variable, the application intelligently directs requests either to the WireMock server (if set to `http://localhost:8080`) or the AccuWeather API. In this scenario, the request is being routed to the WireMock server.
+
+    This code snippet ensures that the `ACCUWEATHER_API_KEY` is only required when the application is not using WireMock, preventing errors and improving efficiency.
```

Comment:
* This feels very GenAI
* Why are we describing the program as part of step 3? I think this should come before. The step list should demonstrate the "how-to", but this and the code excerpt is an explanation.
* We should clarify that the compose service launched in the previous section is a dependency here.

Reply:
I have moved this to Step 3. Highlighted the necessary points and kept it in simple words

Reply:
Also, added a prereq step for WireMock to be up and running .